File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryEventActorsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import com.drogsantoantonio.farmadelivery.data.models.DeliveryEventActorDto
import com.drogsantoantonio.farmadelivery.data.models.DeliveryEventDto
import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryEventActorsTest {
  @Test
  fun `uses the event actor name when present`() {
    assertEquals(
      "Balconista Teste",
      deliveryEventActorLabel(event(actorName = "Balconista Teste")),
    )
  }

  @Test
  fun `trims event actor names`() {
    assertEquals(
      "Motoboy Teste",
      deliveryEventActorLabel(event(actorName = "  Motoboy Teste  ")),
    )
  }

  @Test
  fun `returns null when the event has no actor name`() {
    assertEquals(null, deliveryEventActorLabel(event(actorName = null)))
    assertEquals(null, deliveryEventActorLabel(event(actorName = "   ")))
  }

  private fun event(actorName: String?): DeliveryEventDto {
    return DeliveryEventDto(
      id = "event-1",
      deliveryId = "delivery-1",
      type = "CRIADA",
      actor = actorName?.let {
        DeliveryEventActorDto(
          id = "user-1",
          name = it,
          role = "MOTOBOY",
        )
      },
      createdAt = "2026-05-15T12:00:00.000Z",
    )
  }
}

File: apps/motoboy-pwa/src/deliveryEventActors.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { deliveryEventActorLabel } from "./deliveryEventActors";

test("uses the event actor name when present", () => {
  assert.equal(
    deliveryEventActorLabel({
      actor: {
        id: "user-1",
        name: "Balconista Teste",
        role: "BALCONISTA_CAIXA",
      },
    }),
    "Balconista Teste",
  );
});

test("trims event actor names", () => {
  assert.equal(
    deliveryEventActorLabel({
      actor: {
        id: "user-1",
        name: "  Motoboy Teste  ",
        role: "MOTOBOY",
      },
    }),
    "Motoboy Teste",
  );
});

test("returns null when the event has no actor name", () => {
  assert.equal(deliveryEventActorLabel({ actor: null }), null);
  assert.equal(
    deliveryEventActorLabel({
      actor: {
        id: "user-1",
        name: "   ",
        role: "ADMIN",
      },
    }),
    null,
  );
});
